editor/editor_resource_picker.cpp

```diff
@@ -296,10 +320,20 @@ void EditorResourcePicker::_edit_menu_cbk(int p_which) {
 				propvalues.push_back(p);
 			}
 
-			String orig_type = edited_resource->get_class();
-			Object *inst = ClassDB::instantiate(orig_type);
-			Ref<Resource> unique_resource = Ref<Resource>(Object::cast_to<Resource>(inst));
-			ERR_FAIL_COND(unique_resource.is_null());
+			Ref<Resource> inst;
+			Ref<Script> res_script = edited_resource->get_script();
+			if (res_script.is_valid()) {
+				StringName script_name = ScriptServer::get_global_class_name(res_script->get_path());
+				if (ScriptServer::is_global_class(script_name)) {
+					inst = ScriptServer::instantiate_global_class(script_name);
+				}
+			}
+			if (inst.is_null()) {
+				inst = ClassDB::instantiate(edited_resource->get_class());
+			}
+			ERR_FAIL_COND_MSG(inst.is_null(), "Failed to instantiate resource during Make Unique.");
+			Ref<Resource> unique_resource = Ref<Resource>(inst);
```

Comment:
Sorry, I didn't get that. What do you mean?

Reply:
Not sure which line this was originally referring to, but I think I have an idea of what you were referring to. There are a few places where I think I replaced some string concatenation operations with a more cohesive formatted string, purely because, as a convention, I felt the code was more readable if you had all of the text of the string together in a single c-string, and then have the arguments to be filled in after-the-fact. And I've seen other places in code where they have a lot of formatted string use too, so I believe that is the general preference where applicable.

Reply:
@TechnoPorg Are you referring to what Will says there?

Reply:
No, sorry, that was unclear. I honestly shouldn't have brought this up in the first place, but I meant that `Ref<Resource> unique_resource = Ref<Resource>(inst);` could be replaced by `Ref<Resource> unique_resource = inst;`.